src/main/java/com/coveo/saml/SamlClient.java

```diff
@@ -655,6 +672,15 @@ public void setSPKeys(X509Certificate certificate, PrivateKey privateKey) throws
     spCredential = new BasicX509Credential(certificate, privateKey);
   }
 
+  /**
+   * Set additional service provider credentials.
+   *
+   * @param credentials  additional credentials to be used when decrypting claims
+   */
+  public void setAdditionalSPCredentials(List<Credential> credentials) throws SamlException {
```

Comment:
Saw your comment in the summary, so maybe 
```java
public void addSpCredentials(Credential credential, Credential ...moreCredentials) throws SamlException {
```
to add credentials to the (initially empty) list?

Reply:
I was thinking something like 

`public void addAlternateSPKey(String publicKey, String privateKey) throws SamlException`

So the Credentials would get built in the method, and it would more closely match the call for the main credentials

Reply:
Possibly with a `clearAlternateSPCredentials()` method to empty out the list I guess.

Reply:
That makes sense!
I was thinking about a usage scenario with client code reading additional credentials as X509 entries directly from PKCS12 store, then initialize the SAML client with them. But it is rather a specific case...

Reply:
I've added an equivalent of their support for passing in the Cert and Key instead of file names to the add method in #2 

The more I look at it, the more I think that follows their intent of a simple (to use) library. Unless someone feels strongly otherwise, I'll submit that as our PR to them, and use it for our implementation in the meantime.